pallets/payments/src/lib.rs

```diff
@@ -275,18 +259,17 @@ pub mod pallet {
 		#[pallet::weight(<T as Config>::WeightInfo::pay(remark.as_ref().map(|x| x.len() as u32).unwrap_or(0)))]
 		pub fn pay(
 			origin: OriginFor<T>,
-			beneficiary: AccountIdLookupOf<T>,
+			beneficiary: BeneficiaryOf<T>,
```

Comment:
Is there any reason why we are stepping back from AccountsLookups instead of AccountId? All the frame pallets are relying on AccountsLookups because the conversion capabilities that these type includes.

Reply:
The idea was to allow for a future where beneficiaries are not only accountIds but also something like a membership NFT. `AccountIdLookupOf<T>` usually refers to a `MultiAddress` and the default lookup process is getting the accountId from `MultiAddress::Id(AccountId)` which is usually the only allowed variant.  
I'll leave the `Beneficiary` thing out of the PR for now as this question makes me realize we could later make use of `MultiAddress::Index` to be used for memberships and give them a first class status across the runtime, we will need a custom `Lookup` config in system and then anywhere where a MultiAddress is used users will be able to select a membership(e.g. a balance transfer to a membership with id 123 can be done easily using `MultiAddress:Index(123)` as destination)

Reply:
This sounds reasonable.